gettsim/typing.py

```diff
@@ -17,17 +18,35 @@ def check_if_series_has_internal_type(series, internal_type):
 
     Returns
     -------
-    out : bool
-        Return check variable.
+    out : adjusted pd.Series and boolean
     """
     if internal_type == float:
         out = is_float_dtype(series) or is_integer_dtype(series)
+        if out:
+            return (out, series)
+        else:
+            series = series.astype(float, errors="ignore")
```

Comment:
I would raise an error in this function if type conversion doesn't work out. So I would get rid of `, errors="ignore"` and wrap this line in a try except statement

Structure could be similar to this code snipped from interface.py (you need to find out which kind of Error type conversion would raise, I think ValueError)

    try:
        aggr = agg_specs["aggr"]
    except KeyError:
        raise KeyError(
            f"No aggr keyword is specified for aggregation column {agg_col}."
        )

Also we should specifically check this conversion and potentially raise an error ourselves:
If type is float, conversion to bool or int should fail. But: before all checks are done we should include the following line to make sure that float series that only contain integers are converted to integer:

`series = series.convert_dtypes()`

Reply:
> Also we should specifically check this conversion and potentially raise an error ourselves: If type is float, conversion to bool or int should fail. But: before all checks are done we should include the following line to make sure that float series that only contain integers are converted to integer:
> 
> `series = series.convert_dtypes()`

I am not sure if I fully understand what what you mean here and how it is done

Reply:
I mean we should explicitely check for some combinations of expected type and input type and porentially raise an error:

Something like:

```
if is_integer_dtype(series) and input_type == float:
    raise ValueError("...")